### PR TITLE
Changes to response object

### PR DESCRIPTION
**Pull request by elcorbs**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/PA-399
### *What is the problem we're trying to solve*

Bringing the APIs request/ response objects in line with the new [swagger doc](https://app.swaggerhub.com/apis-docs/Hackney/addresses-api/1.0.0#/default/get_api_v1_addresses)
### *What changes have we introduced*

This is adding a version 2 of the API that we can make the necessary changes on without breaking the API for the current consumers. 
The infrastructure folder has been pulled out of the versioning and is being used by both versions. This is because the infrastructure is what defines the database schema and the database will be consistent across both versions.

#### _Checklist_

- [x] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: AddressesAPI.Tests/V2/Helper/TestDatabaseHelper.cs [hunk 1]
```diff
@@ -0,0 +1,165 @@
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.Data;
+using Microsoft.Data.SqlClient;
```
Comment (Renny4Real) [inline AddressesAPI.Tests/V2/Helper/TestDatabaseHelper.cs:4]: Could we use entity framework here?

File: AddressesAPI/V2/Controllers/GetAddressController.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
+using AddressesAPI.V2.Boundary.Requests;
+using AddressesAPI.V2.Boundary.Responses;
+using AddressesAPI.V2.Boundary.Responses.Metadata;
+using AddressesAPI.V2.UseCase.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+
+namespace AddressesAPI.V2.Controllers
+{
+    [ApiVersion("1")]
```
Comment (Renny4Real) [inline AddressesAPI/V2/Controllers/GetAddressController.cs:9]: Would this need to be `[ApiVersion("2")]`?

File: AddressesAPI/V2/Controllers/GetAddressCrossReferenceController.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
+using AddressesAPI.V2.Boundary.Requests;
+using AddressesAPI.V2.Boundary.Responses;
+using AddressesAPI.V2.Boundary.Responses.Metadata;
+using AddressesAPI.V2.UseCase.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+
+namespace AddressesAPI.V2.Controllers
+{
+    [ApiVersion("1")]
```
Comment (Renny4Real) [inline AddressesAPI/V2/Controllers/GetAddressCrossReferenceController.cs:9]: `[ApiVersion("2")]`?

File: AddressesAPI/V2/Controllers/HealthCheckController.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
+using System.Collections.Generic;
+using AddressesAPI.V2.UseCase;
+using Microsoft.AspNetCore.Mvc;
+
+namespace AddressesAPI.V2.Controllers
+{
+    [ApiVersion("1")]
```
Comment (Renny4Real) [inline AddressesAPI/V2/Controllers/HealthCheckController.cs:7]: `[ApiVersion("2")]` 😄 ?

File: AddressesAPI.Tests/V2/Helper/TestDatabaseHelper.cs (hunk 1 above)
Comment (elcorbs) [reply AddressesAPI.Tests/V2/Helper/TestDatabaseHelper.cs:4]: I think this file is leftover from removing the gateway for the SQL server database. I've removed it in another branch so might leave it here, if that's ok?

File: AddressesAPI/V2/Controllers/SearchAddressController.cs [hunk 5]
```diff
@@ -0,0 +1,64 @@
+using System.Collections.Generic;
+using AddressesAPI.V2.Boundary.Requests;
+using AddressesAPI.V2.Boundary.Responses;
+using AddressesAPI.V2.Boundary.Responses.Metadata;
+using AddressesAPI.V2.UseCase.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+
+namespace AddressesAPI.V2.Controllers
+{
+    [ApiVersion("1")]
```
Comment (Renny4Real) [inline AddressesAPI/V2/Controllers/SearchAddressController.cs:10]: 😁

File: AddressesAPI.Tests/V2/Helper/TestDatabaseHelper.cs (hunk 1 above)
Comment (Renny4Real) [reply AddressesAPI.Tests/V2/Helper/TestDatabaseHelper.cs:4]: 👍

**Review by Renny4Real — APPROVED** [review]
🎉

**Pull request merged**